### PR TITLE
Feature#01 initial setup

### LINKED ISSUE #2: Feature#01_Initial-setup
# REQUIREMENTS
- Set Rails Base Project
- PostgreSQL
- Bootstrap
# Non-functional requirements - Gems 
- RSpect
- RubyCrityc 
- Sandimeter
- Rubocop

# Acceptance Criteria
- can run
- [ ] rubocop
- [ ] rubycritic
- [ ] sandi_meter
- [ ]  rspec

### PR DESCRIPTION
**Pull request by SpidySamurai**
# Description

 - What?: Init rails project. Setup database engine with postgres and install gems and dependencies for the project.
 - Why?: In order to develop with the same environment configurations.  For more background, see ticket Feature #1

 # Non-functional requirements
 Configures rubocop , rubycritic, sandi_meter as code review tools for quality

 # Setup tests
 Configures rspec-rails  for testing.

 # Setup frontend
  Configures bootstrap 5 as frontend framework.

# Preconditions
```
 bundle install & yarn install 
 rails server and bin/webpack-dev-server
```
- Database creation
```
rails  db:create
rails db:migrate
```
# Execute metrics code and sandi_meter
` rake metrics_code`

### CODE REVIEW

**Review by Arkamis — DISMISSED** [review]
LGTM :+1:

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# gem 'image_processing', '~> 1.2'
+
+# Reduces boot times through caching; required in config/boot.rb
+gem 'bootsnap', '>= 1.4.4', require: false
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: %i[mri mingw x64_mingw]
+  gem "rspec-rails", "~> 5.0"
+  gem "sqlite3", "~> 1.4"
```
Comment (anxhe) [inline Gemfile:37]: Perhaps can you remove this gem , I noticed you are not using it in `config/database.yml`

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  gem 'listen', '~> 3.3'
+  gem 'rack-mini-profiler', '~> 2.0'
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem 'rubocop', '~> 1.14'
+  gem 'spring'
+  # Metrics
+  gem 'rake', '~> 13.0'
+  gem 'rubocop-rails', '~> 2.10'
+  gem 'rubycritic', '~> 4.6'
+  gem 'rubycritic-small-badge', '~> 0.2.1'
```
Comment (anxhe) [inline Gemfile:54]: Could it be that you forgot to configure this gem? or are you going to do it later in other issue?

**Review by anxhe — DISMISSED** [review]
You are ready to begin

File: Gemfile (hunk 1 above)
Comment (marianoalarcon) [reply Gemfile:37]: That's right, thanks! in principle we thought to use sqlite3 for development and testing, in this way we keep postgres only for production, but then we choose to keep everything in postgres

**Review by facundotoconas — APPROVED** [review]

**Review by marianoalarcon — APPROVED** [review]

**Review by anxhe — APPROVED** [review]

File: lib/tasks/metrics_code.rake [hunk 3]
```diff
@@ -6,19 +6,24 @@ if Rails.env.development?
...
-    RubyCritic::RakeTask.new("rubycritic:#{dir}") do |task|
-      task.options = %(
-        --custom-format RubyCriticSmallBadge::Report
-        --minimum-score #{RubyCriticSmallBadge.config.minimum_score}
-        --format html
-        --path tmp/rubycritic/#{dir}
-        --format console
-      )
-      task.paths = FileList["#{dir}/**/*.rb"]
+      RubyCritic::RakeTask.new("_rubycritic:#{dir}") do |task|
```
Comment (anxhe) [inline lib/tasks/metrics_code.rake:16]: Nice , I am glad that you liked this approach 😉

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+require 'simplecov_small_badge'
+
+SimpleCov.start(:rails) do
+  SimpleCov.formatters = SimpleCov::Formatter::MultiFormatter.new([
+    SimpleCov::Formatter::HTMLFormatter,
+    SimpleCovSmallBadge::Formatter,
+  ])
+end
+SimpleCov.minimum_coverage(65)
+SimpleCovSmallBadge::Formatter.class_eval do
```
Comment (anxhe) [inline spec/spec_helper.rb:11]: This monkey patch is very useful without this gem fails ,  can see context here: https://github.com/MarcGrimme/simplecov-small-badge/issues/15

**Review by anxhe — APPROVED** [review]
Nice, I see you included several things, it seems that my previous review was gone

File: Gemfile [hunk 5]
```diff
@@ -52,9 +52,13 @@ group :development do
   gem 'rubocop-rails', '~> 2.10'
   gem 'rubycritic', '~> 4.6'
   gem 'rubycritic-small-badge', '~> 0.2.1'
+  gem "hamlit", "~> 2.15"
```
[the remark this answers is not in the document]
Comment (SpidySamurai) [reply Gemfile]: Yes ! This gem improves the dev !

File: spec/spec_helper.rb (hunk 4 above)
Comment (SpidySamurai) [reply spec/spec_helper.rb:11]: Ohh, thank u very much, knowing these kind of stuff helps a lot!

File: lib/tasks/metrics_code.rake (hunk 3 above)
Comment (marianoalarcon) [reply lib/tasks/metrics_code.rake:16]: Yes, we liked the order you keep when making the project settings. We find it efficient and effective

**Pull request merged**